### PR TITLE
add conditional-includes

### PR DESCRIPTION
**Pull request by iamstarkov**

### CODE REVIEW

**Comment by iamstarkov** [comment]
to review https://iamstarkovcom-hnq31i59k.now.sh/conditional-includes/

File: pages/conditional-includes.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+<Meta
+  title="Keeping up with two git identities"
+  description="You can work with your company projects and with open-source projects without mixing two identities."
+/>
+
+# Keeping up with two git identities
+
+_November 13, 2018_
+
+If you are working in the frontend, then you most likely use a lot of open-source projects and perhaps your company even open sourced several internal projects. Also you most likely fix issues you have with 3rd party projects you depend on from time to time.
```
Comment (tipishev) [inline pages/conditional-includes.md]: * "most likely" repeats twice, maybe you could use "chances are" or "probably"
* "3rd party projects you depend on" -> "3rd party dependencies"

File: pages/conditional-includes.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+And everything works fine with you simple global git name and email, but then you start to notice some of your GitHub commits lack an avatar, or quite the opposite in your internal source control system (e.g. hosted BitBucket or Gitlab) your commits lack the avatar and username link is shaded.
```
Comment (tipishev) [inline pages/conditional-includes.md]: * Not sure about starting a sentence with "And"
* "with you" -> "with your"

File: pages/conditional-includes.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+You start to wonder why and after some digging you will find out one of two: either you have been committing to open-source project with your company email or to internal projects with your private email. None of these two options are good.
```
Comment (tipishev) [inline pages/conditional-includes.md]: You start to wonder why and after some digging you find out that you committed to an open-source project using your company email, or the other way around.

File: pages/conditional-includes.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    ├── mdx
+    └── next.js
+```
+
+Then you want to use different identities for different purposes. Conditional includes will help with that.
+
+> You can include a config file from another conditionally by setting a `includeIf.<condition>.path` variable to the name of the file to be included.  
+> [git docs](https://git-scm.com/docs/git-config#_conditional_includes)
+
+In order to utilise this feature you will need three files:
```
Comment (tipishev) [inline pages/conditional-includes.md]: ~~will~~

File: pages/conditional-includes.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+- `~/.gitconfig`
+- `~/.gitconfig_oss`
+- `~/.gitconfig_company`
+
+And content of these files will look like this:
```
Comment (tipishev) [inline pages/conditional-includes.md]: will -> should

File: pages/conditional-includes.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+$ cat ~/.gitconfig_company
+[user]
+  email = Vladimir.Starkov@company.org
+```
+
+Let's go through what is going on here.
+
+in `~/.gitconfig` you specify general options which can be overridden by subconfigs. In this case you specify `user.name` option and it is `Vladimir Starkov` in my case. After that you have two `includeIf` sections, with `gitdir` option set to relevant folders with paths set to relevant subconfigs.
+Which in turn specify different `user.email` values for different purposes: in my case it is `user.email=iamstarkov@gmail.com` for open-source projects and `user.email=Vladimir.Starkov@company.org`. It was fast, huh?
```
Comment (tipishev) [inline pages/conditional-includes.md]: maybe use bullet-points?
---
* `user.email=iamstarkov@gmail.com` for open-source
* `user.email=Vladimir.Starkov@company.org`  for company.org projects

File: pages/conditional-includes.md (hunk 2 above)
Comment (mjmonline) [inline pages/conditional-includes.md]: Maybe something along the line with: While everything may seem to be working fine with your simple global git name and email, you then start to notice...

File: pages/conditional-includes.md (hunk 1 above)
Comment (iamstarkov) [reply pages/conditional-includes.md]: fixed

File: pages/conditional-includes.md (hunk 2 above)
Comment (iamstarkov) [reply pages/conditional-includes.md]: fixed

File: pages/conditional-includes.md (hunk 3 above)
Comment (iamstarkov) [reply pages/conditional-includes.md]: fixed

File: pages/conditional-includes.md (hunk 4 above)
Comment (iamstarkov) [reply pages/conditional-includes.md]: fixed

File: pages/conditional-includes.md (hunk 5 above)
Comment (iamstarkov) [reply pages/conditional-includes.md]: fixed

File: pages/conditional-includes.md (hunk 6 above)
Comment (iamstarkov) [reply pages/conditional-includes.md]: nah

File: pages/conditional-includes.md (hunk 2 above)
Comment (iamstarkov) [reply pages/conditional-includes.md]: looks better

**Pull request merged**